flume-ng-sinks/flume-ng-kafka-sink/src/test/java/org/apache/flume/sink/kafka/TestKafkaSink.java

```diff
@@ -239,6 +240,42 @@ public void testTopicAndKeyFromHeader() throws UnsupportedEncodingException {
                  new String((byte[]) fetchedMsg.key(), "UTF-8"));
   }
 
+  @Test
+  public void testReplaceSubStringOfTopicWithHeaders() throws UnsupportedEncodingException {
+    Sink kafkaSink = new KafkaSink();
+    Context context = prepareDefaultContext();
+    context.put(TOPIC_CONFIG, TestConstants.HEADER_TOPIC);
+    Configurables.configure(kafkaSink, context);
+    Channel memoryChannel = new MemoryChannel();
+    Configurables.configure(memoryChannel, context);
+    kafkaSink.setChannel(memoryChannel);
+    kafkaSink.start();
+
+    String msg = "test-replace-substring-of-topic-with-headers";
+    Map<String, String> headers = new HashMap<String, String>();
```

Comment:
Nit: explicit type argument can be replaced by `<>`. Or `Collections.singletonMap()` might be even better.